### PR TITLE
Update iresearch to fix bugs

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

* Fix SEARCH-467
* Fix SEARCH-473
* Fix SEARCH-474

Should be in 3.11!

* Store all posting list max score
* Store delta between freq and norm instead of just norm in wand data

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/IResearch/IResearchFeature.cpp [hunk 1]
```diff
@@ -114,50 +115,42 @@ class IResearchLogTopic final : public LogTopic {
...
-    irsLevel = std::min(irsLevel, irs::logger::IRL_TRACE);
-    irs::logger::output_le(irsLevel, log_appender, nullptr);
-  }
+template<LogLevel Level>
+static void log(irs::SourceLocation&& source, std::string_view message) {
+  Logger::log("9afd3", source.func.data(), source.file.data(),
+              static_cast<int>(source.line), Level, LIBIRESEARCH.id(), message);
+}
+
+static constexpr irs::log::Callback kLogs[6] = {
```
Comment (Dronplane) [inline arangod/IResearch/IResearchFeature.cpp:137]: std::array<Callback, 6> ?

Comment (Dronplane) [reply arangod/IResearch/IResearchFeature.cpp:137]: At least to have size and do not write constant "6" several times

Comment (MBkkt) [reply arangod/IResearch/IResearchFeature.cpp:137]: ok, but later or if this jenkins run will failed

Comment (MBkkt) [reply arangod/IResearch/IResearchFeature.cpp]: Fixed

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**